### ISSUE TITLE
[Bug]: false positive with package.json

### ISSUE THREAD

**Issue by Zamiell** [streetsidesoftware/vscode-spell-checker#2864]
## Description

The CSpell VSCode extension shows a blue squiggly line under the word "heroicons" inside of a "package.json" file.

However, when running CSpell from command line, no errors are triggered.

This is a bug because I expect the output of the VSCode extension to exactly match that of the command line tool.

## Steps to Reproduce

```
git clone git@github.com:Zamiell/isaac-streaking.git
cd isaac-streaking
git checkout caf2be00ec8da6f211f746ed65be26cd0d476184
yarn install
npx cspell package.json # Notice that there are no errors.
code .
```

Then, open "package.json" in VSCode and observe that there is a blue squiggly underline.

**Comment by Jason3S** [comment]
@Zamiell,

I was not able to reproduce this. 

But, it might be related to #2666. I hope #2868 will help with this sort of issue.

**Comment by Zamiell** [comment]
Here's another reproduction:

```
git clone git@github.com:Zamiell/isaac-among-us.git
cd isaac-among-us
yarn install
npx cspell --no-progress . # Notice that there are no errors.
code .
```

Then, do Ctrl + P, open "SabotageType.ts" in VSCode and observe that there is a blue squiggly underline.

**Comment by Jason3S** [comment]
@Zamiell,

Do you mean `comms`?

<img width="235" alt="image" src="https://github.com/streetsidesoftware/vscode-spell-checker/assets/3740137/a6b3f616-cb3e-4989-ab3c-ea8198a9ad09">

That is because comms was recently added to the dictionary and the extension has not yet been updated.

**Comment by Zamiell** [comment]
Ah. Shouldn't the two be updated at the same time to prevent unfixable errors like this?

**Comment by Zamiell** [comment]
@Jason3S Can  you respond to my previous comment?

Additionally, I found the same bug yet again with another trivial reproduction:

```
mkdir foo
cd foo
npm init --yes
npm install --save cspell clsx
npx cspell package.json # Notice that there are 0 errors.
code . # Notice that if you open "package.json", there is a blue squiggly line in VSCode.
```

Please let me know if this reproduction works for you.

**Comment by Jason3S** [comment]
@Zamiell,

I try to keep the code extension up to date with the `cspell` releases, but it is going though a major update at the moment, hence a delayed release.

Given that, if `cspell` is installed in the repository, the extension should pick up the latest dictionaries. Then the results would match as long as no new dictionaries were added.

**Comment by Jason3S** [comment]
> @Jason3S Can you respond to my previous comment?
> 
> Additionally, I found the same bug yet again with a trivial reproduction that does not involve cloning any existing repositories:
> 
> ```
> mkdir foo
> cd foo
> npm init --yes
> npm install --save cspell clsx
> npx cspell package.json # Notice that there are 0 errors.
> code . # Notice that if you open "package.json", there is a blue squiggly line in VSCode.
> ```
> 
> Please let me know if this reproduction works for you.

I'll try it out.

**Comment by Zamiell** [comment]
Awesome, thanks Jason. Thank you for your work on CSpell; I eagerly await version 4.0!
